src/shop-payment-request.html

```diff
@@ -0,0 +1,254 @@
+<!--
+@license
+Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../bower_components/polymer/polymer.html">
+
+<dom-module id="shop-payment-request">
+  <script>
+
+    Polymer({
+      is: 'shop-payment-request',
+
+      properties: {
+        currency: {
+          type: String,
+          value: 'USD'
+        },
+
+        supportedMethods: {
+          type: Array,
+          value: function() { return ["visa", "mastercard", "amex", "discover", "maestro", "diners", "jcb", "unionpay", "bitcoin"]; }
+        },
+
+        requestPayerPhone: {
+          type: Boolean,
+          value: false
+        },
+
+        requestPayerEmail: {
+          type: Boolean,
+          value: false
+        },
+
+        androidPay: {
+          type: Object,
+          value: function() { return {}; }
+        },
+
+        summaryLabel: {
+          type: String,
+          value: "Total"
+        },
+
+        _paymentTypesSupported: {
+          computed: '_computePaymentTypesSupported(supportedMethods, androidPay)'
+        }
+      },
+
+      request: function(itemDetails, totalCost, currency, clearCartOnSuccess) {
+        if (!'PaymentRequest' in window) {
+          this._redirectToCheckout();
```

Comment:
nit: could we fire the `error` event? That way, we could simplify this operation to a success/error message, then `_onError` could redirect depending on the status code.  
A proposal:
1. Remove `_redirectToCheckout`
2. Fire `error` and specify an error_code.

Also, this method could return a promise in the *future*.